File: ArdavanGhaffari/QL/src/main/java/ast/model/statement/IfStatement.java
package ast.model.statement;

import ast.model.Box;
import ast.model.Expression;
import ast.model.Statement;
import ast.visitor.StatementVisitor;

public class IfStatement extends Statement {
	private Expression expression;
	private Box ifBox;
	
	public IfStatement(Expression expression, Box box) {
		this.expression = expression;
		this.ifBox = box;
	}

	public Expression getExpression() {
		return expression;
	}

	public Box getIfBox() {
		return ifBox;
	}

	@Override
	public void accept(StatementVisitor statementVisitor) {
		statementVisitor.visit(this);
	}
	
}

File: JordyHeemskerk/QL/QL/IfStatement.swift
//
//  IfStatement.swift
//  QL
//
//  Created by Jordy Heemskerk on 24/02/16.
//  Copyright © 2016 Jordy Heemskerk. All rights reserved.
//

import Foundation
import SwiftParsec

struct IfStatement: Statement {
    
    let conditionClause: Expression
    let block: Block
    let elseClause: ElseIfStatement?
    let position: Position
    
    init(tupleInput: (((Expression, Block), [ElseIfStatement]), Position)) {
        let (((conditionClause, block), elseClause), currentPosition) = tupleInput
        self.conditionClause = conditionClause
        self.block = block
        self.elseClause = elseClause.first
        self.position = currentPosition
    }
    
    func accept(visitor: ASTVisitor) {
        visitor.visit(self)
    }
}